crates/rbuilder/src/building/sim.rs

```diff
@@ -434,8 +437,8 @@ pub fn simulate_order(
 
 /// Simulates order (including parent (those needed to reach proper nonces) orders) using a precreated fork
 pub fn simulate_order_using_fork<Tracer: SimulationTracer>(
-    parent_orders: Vec<Order>,
-    order: Order,
+    parent_orders: &[Order],
```

Comment:
using the reference allows us to not clone the parents, saving a potentially _very_ large allocation